core/src/apps/solana/layout.py

```diff
@@ -111,14 +113,19 @@ async def confirm_instruction(
 
             account_data: list[tuple[str, str]] = []
             if len(account_value) == 2:
+                token_suffix = ""
+                if account_template.is_token_id and definitions:
```

Comment:
I think that it can't ever happen that both `token_suffix` and `signer_suffix` will have some value. That's why I think it'd be better (more readable) to have a single `token_suffix` variable which would include either the ticker or `signer_suffix` and we'd use a `if/else` statement here instead of two `if`s. WDYT?